### PR TITLE
Feature/burst service

### PR DESCRIPTION
**Pull request by kforeverisback**
# Type of PR

- [ ] Documentation changes
- [X] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR

Early implementation of burst service with K8S HPA API.
Couple more improvements could be added but primary functionality (k8s HPA api call) is implemented.

## Does this introduce a breaking change

- [ ] YES
- [X] NO

## Validation

- [X] Ran integration tests
- [ ] Update documentation or issue referenced above

## Issues Closed or Referenced

** Not Sure if it closes https://github.com/retaildevcrews/wcnp/issues/259 **

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,14 +20,14 @@ create : delete build
 
 	kubectl wait node --for condition=ready --all --timeout=60s
 
-	# Install prometheus
-	#@kubectl apply -f ${ISTIO_HOME}/samples/addons/prometheus.yaml
```
Comment (bartr) [inline Makefile:24]: can you double check this?  Makefile is really particular about tabs vs. spaces and it looks like you replaced a tab with 4 spaces.

Comment (kforeverisback) [reply Makefile:24]: I only did this to remove printing of commented shell commands. TAB (or shell recipes) get printed (if not preceded with @).

Should I re-replace them with tabs?

File: burst/Controllers/ApiController.cs [hunk 2]
```diff
@@ -4,26 +4,49 @@
...
+            }
         }
 
         [HttpGet("{ns}/{deployment}")]
-
         public IActionResult Get(string ns, string deployment)
         {
-            Console.WriteLine($"{DateTime.Now:s}  {Request.Path.ToString()}");
-            return Ok($"service: {ns}/{deployment}, current-load: 27, target-load: 60, max-load: 85");
+            var hpaMetrics = _service.GetK8SHPAMetrics(ns, deployment);
```
Comment (bartr) [inline burst/Controllers/ApiController.cs:38]: don't use weakly typed var

File: burst/K8sApi/K8sMetricsService.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+        public Task StopAsync(CancellationToken stoppingToken)
+        {
+            _logger.LogInformation("Stopping k8s HPA-API Service...");
+            _timer?.Change(Timeout.Infinite, 0);
+
+            return Task.CompletedTask;
+        }
+
+        public void Dispose()
```
Comment (bartr) [inline burst/K8sApi/K8sMetricsService.cs:157]: use the disposing pattern

File: burst/Startup.cs [hunk 4]
```diff
@@ -25,7 +26,10 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-
+            services.AddSingleton<IK8sHPAMetricsService, K8sHPAMetricsService>();
```
Comment (bartr) [inline burst/Startup.cs:29]: Why use DI vs a static singleton?

File: burst/K8sApi/K8sHPAMetrics.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using k8s;
+using k8s.Models;
+
+namespace Burst.K8sApi
+{
+    public class K8sHPAMetrics
```
Comment (bartr) [inline burst/K8sApi/K8sHPAMetrics.cs:13]: why nullable?

File: burst/Controllers/ApiController.cs [hunk 6]
```diff
@@ -4,26 +4,49 @@
...
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Hosting;
 
 namespace Burst.Controllers
 {
+    using Burst.K8sApi;
+    using System.Threading;
+
     [ApiController]
     [Route("/burstmetrics")]
```
Comment (bartr) [inline burst/Controllers/ApiController.cs:15]: general convention is for [Controller] to be the first part of the path - rename class to BurstMetricsController?

File: burst/K8sApi/K8sMetricsService.cs [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+namespace Burst.K8sApi
+{
+    public interface IK8sHPAMetricsService
+    {
+        K8sHPAMetrics GetK8SHPAMetrics(string ns, string deployment);
+    }
+    public class K8sHPAMetricsService : IHostedService, IDisposable, IK8sHPAMetricsService
+    {
+        private readonly ILogger<K8sHPAMetricsService> _logger;
```
Comment (bartr) [inline burst/K8sApi/K8sMetricsService.cs:19]: need to lint - _ in member names is an error

File: burst/K8sApi/K8sMetricsService.cs [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+                {
+                    // We're interested in CPU metrics
+                    if (m.Resource.Name == "cpu")
+                    {
+                        return m.Resource.Target.AverageUtilization;
+                    }
+                }
+            }
+
+            _logger.LogWarning("HPA Spec is not set");
```
Comment (bartr) [inline burst/K8sApi/K8sMetricsService.cs:79]: this is not necessarily true - hpa could be null

**Pull request merged**